### PR TITLE
asleap: init at 06-20-2021

### PR DESCRIPTION
**Pull request by pyrox0**
###### Description of changes
One of the packages mentioned in #81418

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/asleap/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{lib, stdenv, fetchFromGitHub, openssl, libpcap, libxcrypt}:
+
+stdenv.mkDerivation rec {
+  pname = "asleap";
+  version = "06-20-2021";
```
Comment (fabaff) [inline pkgs/tools/networking/asleap/default.nix:5]:
```suggestion
  version = "unstable-2021-06-20";
```

File: pkgs/tools/networking/asleap/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    install -Dm644 asleap $out/bin/asleap
+    install -Dm644 genkeys $out/bin/genkeys
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/zackw/asleap";
+    description = "Recovers weak LEAP and PPTP passwords";
+    license = licenses.gpl2Only;
+    platforms = platforms.all;
```
Comment (fabaff) [inline pkgs/tools/networking/asleap/default.nix:25]: Should be the default already.

```suggestion
```

File: pkgs/tools/networking/asleap/default.nix (hunk 1 above)
Comment (ModdedGamers) [reply pkgs/tools/networking/asleap/default.nix]: Fixed.

File: pkgs/tools/networking/asleap/default.nix (hunk 2 above)
Comment (ModdedGamers) [reply pkgs/tools/networking/asleap/default.nix]: Fixed.

**Comment by fabaff** [comment]
@ofborg eval

**Comment by fabaff** [comment]
Error:

```
error: undefined variable 'thehedgeh0g'

       at /home/fab/.cache/nixpkgs-review/pr-175969/nixpkgs/pkgs/tools/networking/asleap/default.nix:25:39:

           24|     license = licenses.gpl2Only;
           25|     maintainers = with maintainers; [ thehedgeh0g ];
             |                                       ^
           26|   };
```

**Comment by ModdedGamers** [comment]
Should be fixed now that #176728 is merged.

File: pkgs/tools/networking/asleap/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    repo = pname;
+    rev = "eb3bd42098cba42b65f499c9d8c73d890861b94f";
+    sha256 = "sha256-S6jS0cg9tHSfmP6VHyISkXJxczhPx3HDdxT46c+YmE8=";
+  };
+
+  buildInputs = [ openssl libpcap libxcrypt ];
+
+  installPhase = ''
+    install -Dm644 asleap $out/bin/asleap
+    install -Dm644 genkeys $out/bin/genkeys
```
[the remark this answers is not in the document]
Comment (theHedgehog0) [reply pkgs/tools/networking/asleap/default.nix]: Should be fixed. Thought I had already pushed my changes. Whoops!

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 175969` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>asleap</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/175969#issuecomment-1156960225)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-175969]$ ./results/asleap/bin/asleap 
asleap 2.3 - actively recover LEAP/PPTP passwords. <jwright@hasborg.com>
asleap: Must supply an interface with -i, or a stored file with -r
Usage: asleap [options]
[...]
```

Looks good to me.

**Pull request merged**